profiling-ffi/src/crashtracker/crash_info.rs

```diff
@@ -0,0 +1,214 @@
+///
+/// # Safety
+/// `crashinfo` must be a valid pointer to a `CrashInfo` object.
+/// All references inside `metadata` must be valid.
+/// Strings are copied into the crashinfo, and do not need to outlive this call.
+#[no_mangle]
+#[must_use]
+pub unsafe extern "C" fn ddog_crashinfo_set_siginfo(
+    crashinfo: *mut CrashInfo,
+    siginfo: SigInfo,
+) -> CrashtrackerResult {
+    (|| {
+        let crashinfo = crashinfo_ptr_to_inner(crashinfo)?;
+        let siginfo = siginfo.try_into()?;
+        crashinfo.set_siginfo(siginfo)
+    })()
+    .context("ddog_crashinfo_set_siginfo failed")
+    .into()
+}
+
+/// If `thread_id` is empty, sets `stacktrace` as the default stacktrace.
+/// Otherwise, adds an additional stacktrace with id "thread_id".
+///
+/// # Safety
+/// `crashinfo` must be a valid pointer to a `CrashInfo` object.
+/// All references inside `stacktraces` must be valid.
+/// Strings are copied into the crashinfo, and do not need to outlive this call.
+#[no_mangle]
+#[must_use]
+pub unsafe extern "C" fn ddog_crashinfo_set_stacktrace(
+    crashinfo: *mut CrashInfo,
+    thread_id: CharSlice,
+    stacktrace: Slice<StackFrame>,
+) -> CrashtrackerResult {
+    (|| {
+        let crashinfo = crashinfo_ptr_to_inner(crashinfo)?;
+        let thread_id = option_from_char_slice(thread_id)?;
+        let mut stacktrace_vec = Vec::with_capacity(stacktrace.len());
+        for s in stacktrace.iter() {
+            stacktrace_vec.push(s.try_into()?)
+        }
+        crashinfo.set_stacktrace(thread_id, stacktrace_vec)
+    })()
+    .context("ddog_crashinfo_set_metadata failed")
+    .into()
+}
+
+/// Exports `crashinfo` to the Instrumentation Telemetry backend
+///
+/// # Safety
+/// `crashinfo` must be a valid pointer to a `CrashInfo` object.
+#[no_mangle]
+#[must_use]
+pub unsafe extern "C" fn ddog_crashinfo_upload_to_telemetry(
+    crashinfo: *mut CrashInfo,
+    config: CrashtrackerConfiguration,
+) -> CrashtrackerResult {
+    (|| {
+        let crashinfo = crashinfo_ptr_to_inner(crashinfo)?;
+        let config = config.try_into()?;
+        crashinfo.upload_to_telemetry(&config)
+    })()
+    .context("ddog_crashinfo_upload_to_telemetry failed")
+    .into()
+}
+
+/// Exports `crashinfo` to the profiling backend at `endpoint`
+/// Note that we support the "file://" endpoint for local file output.
+/// # Safety
+/// `crashinfo` must be a valid pointer to a `CrashInfo` object.
+#[no_mangle]
+#[must_use]
+pub unsafe extern "C" fn ddog_crashinfo_upload_to_endpoint(
+    crashinfo: *mut CrashInfo,
+    endpoint: ProfilingEndpoint,
+    timeout_secs: u64,
+) -> CrashtrackerResult {
+    (|| {
+        let crashinfo = crashinfo_ptr_to_inner(crashinfo)?;
+        let endpoint = exporter::try_to_endpoint(endpoint)?;
+        let timeout = Duration::from_secs(timeout_secs);
+        crashinfo.upload_to_endpoint(endpoint, timeout)?;
+        anyhow::Ok(())
+    })()
+    .context("ddog_crashinfo_upload_to_endpoint failed")
+    .into()
+}
```

Comment:
This was somewhat unexpected to me. Shouldn't libdatatadog be the one in control of what gets sent where? It's weird to hand this over to the API user... I think?